### PR TITLE
The secret can be any CipherKey, not just a String.

### PR DESCRIPTION
**Pull request by jyasskin**

### CODE REVIEW

**Comment by natevw** [comment]
Thanks, looks like this is a necessary comment-based documentation update for the work done on #33.

I'm a bit unclear on using `crypto.CipherKey` to describe the field though. Is that your own shorthand? The current docs for https://nodejs.org/api/crypto.html#cryptocreatehmacalgorithm-key-options don't use that terminology.

(They do mention something called `CryptoKey` but that's just one of the options and afaict is a [specific type of object](https://developer.mozilla.org/en-US/docs/Web/API/CryptoKey) in the Web Crypto API.)

Afaict there's no official type definitions for node.js, but my IDE relies on [@types/node](https://www.npmjs.com/package/@types/node) as I suspect many others do, and their [current definition for `createHmac`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/28e2ef6/types/node/crypto.d.ts#L278) uses `BinaryLike | KeyObject`. Maybe we can borrow those names as being close enough human-readable descriptions too?

File: index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var crypto = require('crypto');
  * Sign the given `val` with `secret`.
  *
  * @param {String} val
- * @param {String} secret
+ * @param {crypto.CipherKey} secret
```
Comment (natevw) [inline index.js:11]: see background in the PR comments, but maybe `@param {String|BinaryLike|KeyObject} secret` would be most clear here?

[technically "String" is already part of "BinaryLike" in the typescript defs that we're borrowing from, but I'd rather err on the side of people knowing a string is still fine here]

File: index.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ exports.sign = function(val, secret){
  * returning `false` if the signature is invalid.
  *
  * @param {String} input
- * @param {String} secret
+ * @param {crypto.CipherKey} secret
```
Comment (natevw) [inline index.js:31]: [same here as whatever settled on above]

**Comment by jyasskin** [comment]
Ah, sorry for not explaining that. `CipherKey` is defined at https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0039ca2aaaf3b372e9664fd2e765a1eba3ca5a46/types/node/crypto.d.ts#L675 as
```typescript
type CipherKey = BinaryLike | KeyObject;
```
so just a shorthand for the argument type of `createHmac`. And in fact, `tsc *.js --checkJs --noEmit` passes with this change.

+1 that including "String" is probably clearer for JS folks trying to refer to the nodejs docs. It might also be clearest to expand out `BinaryLike` to `string | NodeJS.ArrayBufferView`. I've done that, but let me know if you'd prefer to use BinaryLike instead.

**Comment by natevw** [comment]
Whoops, sorry I missed your updates here! Looks good to me, I'll merge and bump a patch release for this.

**Comment by natevw** [comment]
Published as `cookie-signature@1.2.1` — thanks again for getting this caught up!

**Pull request merged**